argilla/src/argilla/records/_dataset_records.py

```diff
@@ -398,92 +410,103 @@ def _validate_vector_names(self, vector_names: Union[List[str], str]) -> None:
             if vector_name not in self.__dataset.schema:
                 raise ValueError(f"Vector field {vector_name} not found in dataset schema.")
 
+    def _render_record_mapping(
+        self,
+        records: List[Dict[str, Any]],
+        mapping: Optional[Dict[str, Union[str, List[str], Tuple[str]]]] = None,
+    ) -> Dict[str, Tuple[Optional[str]]]:
+        """Renders a mapping from a list of records and a mapping dictionary, to a singular mapping dictionary."""
+        schema = self.__dataset.schema
+        mapping = mapping or {}
+        singular_mapping = defaultdict(dict)
+
+        # update the mapping with unmapped columns
+        for key, value in records[0].items():
+            if key not in schema and key not in mapping:
+                warnings.warn(
```

Comment:
not sure if we would skip in both cases? I think if it is not in the schema, I would say the user made a mistake right? If it it not present in the one record, I would assume it might by chance have an empty value.

Reply:
If a value is not present in the schema or mapping, then we assume that it's something that the user does not want to log to Argilla. We warn in case they've made a mistake.